library/src/scala/quoted/show/showName.scala

```diff
@@ -0,0 +1,24 @@
+package scala.quoted.show
+
+/** Annotation used inside a quote to give a custom name to a definition.
+ *  The `name` argument must be a literal String.
+ *
+ *  Usage:
+ *  ```scala
+ *  def let(name: String)(value: Expr[Int])(in: Expr[Int] => Expr[Int]): Expr[Int] = '{
+ *    @showName(${Expr(name)})
+ *    val x = $value
+ *    ${ in('x) }
+ *  }
+ *  ```
+ *  then using it in
+ *  ```scala
+ *  let("myVal")('{4})(x => '{ $x + 1}).show
+ *  ```
+ *  will retuns the code
+ *  ```scala
+ *  val myVal = 4
+ *  myVal + 1
+ *  ```
+ */
+class showName(name: String) extends scala.annotation.Annotation
```

Comment:
Minor: what about `displayName` or `debugName`? `showName` feels more like a verb.

Reply:
I chose show name because only the `Expr.show` will display it. Compiler debugging options  as `-Xprint` will just print the annotations.